third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -115,6 +129,110 @@ struct LoadStoreConversionBase {
     return std::min<unsigned>(128 / pointeeBitWidth, contiguity);
   }
 
+  // Look through the available assumption to verify if the expression has been
```

Comment:
Yeah I spoke with @antiagainst just now and yes, I can pull it out so that we have a flow like:
- A ttgir pass (probably after canonicalization) that converts tt.load in tt.amdgpu.buffer_load
- A new LLVM conversion pass that lowers the buffer load to LLVM

Reply:
Yeah let's define the buffer load op first and connect. We can rebase this pull request later to pick it up. We may not need a new LLVM conversion pass; just op lowering patterns to pull into existing conversion pass.

Reply:
I was thinking:
 - 1 PR with the `compiler.py` fix (+test that @ThomasRaoux  was proposing)
 - 1 PR with the buffer load/store ops + lowering
 - 1 PR with the conversion from tt.load to buffer_load

Reply:
Hi @ThomasRaoux @antiagainst , I think I hit a roadblock on point number 2 (i.e., introducing the `amdgpu.buffer_load`). In the following IR:
```
#blocked0 = #triton_gpu.blocked<{sizePerThread = [8], threadsPerWarp = [32], warpsPerCTA = [1], order = [0], CTAsPerCGA = [1], CTASplitNum = [1], CTAOrder = [0]}>
module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32} {
    // CHECK-LABEL: buffer_load_store_vec2
    tt.func @buffer_load_store_vec2(%arg0: !tt.ptr<f16> {tt.divisibility = 4 : i32}) {
        %c256_i32 = arith.constant 256 : i32
        %0 = tt.get_program_id x : i32
        %1 = arith.muli %0, %c256_i32 : i32
        %2 = tt.make_range {end = 256 : i32, start = 0 : i32} : tensor<256xi32, #blocked0>
        %3 = tt.splat %1 : i32 -> tensor<256xi32, #blocked0>
        %4 = arith.addi %3, %2 : tensor<256xi32, #blocked0>
        %9 = amdgpu.buffer_load %arg0[%4]: tensor<256xf16, #blocked0>
        tt.return
   }
}
```
How can I ensure that the `tt.divisibility` is correctly picked up by the `buffer_load` operation in `%9` ? Should I add `amdgpu.buffer_load` to the `axisInfo` analysis ?

Reply:
Ah, maybe I sorted this out. I created an internal `getContiguity(ptr, offset)`  where I do not consider the `contiguity` of the base pointer (since it's a scalar, it'll always be 1).

Reply:
Sounds like you figured it out. `divisibility` should still be correct for scalar and you should be able to calculate the contiguity of the access without adding `buffer_load` to the analysis.